pkg/operation/botanist/component/extensions/dnsrecord/dnsrecord.go

```diff
@@ -223,6 +223,14 @@ func (c *dnsRecord) Migrate(ctx context.Context) error {
 
 // Destroy deletes the DNSRecord resource.
 func (c *dnsRecord) Destroy(ctx context.Context) error {
+	if _, err := controllerutils.GetAndCreateOrMergePatch(ctx, c.client, c.secret, func() error {
+		c.secret.Type = corev1.SecretTypeOpaque
+		c.secret.Data = c.values.SecretData
+		return nil
+	}); err != nil {
+		return err
+	}
```

Comment:
This seems to be the same code like in `Deploy`. Should we extract this in a dedicated function?